src/sentry/static/sentry/app/components/confirm.tsx

```diff
@@ -14,268 +14,259 @@ type MessageRenderProps = {
    * Closes the modal, if `bypass` is true, will call `onConfirm` callback
    */
   close: (e: React.MouseEvent) => void;
-};
-
-type ChildrenRenderProps = {
-  open: () => void;
-  close: () => void;
-};
-
-const defaultProps = {
   /**
-   * Button priority
-   */
-  priority: 'primary' as React.ComponentProps<typeof Button>['priority'],
-  /**
-   * Disables the confirm button
+   * Set the disabled state of the confirm button
    */
-  disableConfirmButton: false,
+  disableConfirmButton: (disable: boolean) => void;
   /**
-   * Text to show in the cancel button
+   * When the modal is confirmed the function registered in here weill be
+   * called.
+   *
+   * Usefull if your rendered message contains some functioanlity that should
+   * be triggered upon the modal being confirmed.
    */
-  cancelText: t('Cancel') as React.ReactNode,
-  /**
-   * Text to show in the confirmation button
-   */
-  confirmText: t('Confirm') as React.ReactNode,
-  // Stop event propagation when opening the confirm modal
-  stopPropagation: false,
+  setConfirmCallback: (cb: () => void) => void;
```

Comment:
wouldn't it be enough to just use the onConfirm prop? do we need this functionality atm?

Reply:
The comment above describes the usecase.

If you're rendering content within the modal the parent component which is rendering the `<Confirm` will not have access to the state of things rendered inside of the modal because of how our global modals work.